src/frontend/src/icp/workers/icrc-wallet.worker.ts

```diff
@@ -61,33 +64,73 @@ const mapTransaction = ({
 	return mapIcrcTransaction({ transaction, identity });
 };
 
+const getBalance = ({
+	identity,
+	certified,
+	data
+}: SchedulerJobParams<PostMessageDataRequestIcrc>): Promise<bigint> => {
+	assertNonNullish(data, 'No data - ledgerIndexCanister - provided to fetch balance.');
+
+	return balance({
+		identity,
+		certified,
+		owner: identity.getPrincipal(),
+		...data
+	});
+};
+
+const MSG_SYNC_ICRC_WALLET = 'syncIcrcWallet';
+
 // Exposed for test purposes
-export const initIcrcWalletScheduler = (): IcWalletTransactionsScheduler<
+export const initIcrcWalletTransactionsScheduler = (): IcWalletTransactionsScheduler<
 	IcrcTransaction,
 	IcrcTransactionWithId,
-	PostMessageDataRequestIcrc
+	PostMessageDataRequestIcrcStrict
 > =>
 	new IcWalletTransactionsScheduler(
 		getTransactions,
 		mapTransactionIcrcToSelf,
 		mapTransaction,
-		'syncIcrcWallet'
+		MSG_SYNC_ICRC_WALLET
 	);
 
-const scheduler = initIcrcWalletScheduler();
+// Exposed for test purposes
+export const initIcrcWalletBalanceScheduler =
+	(): IcWalletBalanceScheduler<PostMessageDataRequestIcrc> =>
+		new IcWalletBalanceScheduler(getBalance, MSG_SYNC_ICRC_WALLET);
+
+const initScheduler = (
+	data: PostMessageDataRequestIcrc | undefined
+): IcWalletScheduler<PostMessageDataRequestIcrc> => {
+	const { success: withIndexCanister } = PostMessageDataRequestIcrcStrictSchema.safeParse(data);
+
+	return withIndexCanister
+		? initIcrcWalletTransactionsScheduler()
+		: initIcrcWalletBalanceScheduler();
+};
+
+let scheduler: IcWalletScheduler<PostMessageDataRequestIcrc> | undefined;
 
 onmessage = async ({ data: dataMsg }: MessageEvent<PostMessage<PostMessageDataRequestIcrc>>) => {
 	const { msg, data } = dataMsg;
 
 	switch (msg) {
-		case 'stopIcrcWalletTimer':
-			scheduler.stop();
-			return;
 		case 'startIcrcWalletTimer':
```

Comment:
Yes, I want to stop the scheduler before starting it BUT, tthe important part here is that `initScheduler` either initialize a scheduler with or without index. So technically, it's possible that a start leads to intializing a new type of scheduler.

Regarding merging the two switch, I prefer the current implementation because that way we avoid duplicating the line of code `scheduler.stop()`.

Reply:
perfect! thank you for confirming!